drivers/tests/TESTS/mbed_drivers/buffered_serial/CMakeLists.txt

```diff
@@ -3,11 +3,20 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+set(MBED_TEST_LINK_LIBRARIES "mbed-os" CACHE STRING "")
+string(FIND "${MBED_TEST_LINK_LIBRARIES}" "mbed-baremetal" string_found_position)
+if(${string_found_position} LESS 0)
+    string(FIND "${MBED_TEST_LINK_LIBRARIES}" "mbed-os" string_found_position)
+    if(${string_found_position} LESS 0)
+        string(APPEND MBED_TEST_LINK_LIBRARIES "mbed-os")
+    endif()
+endif()
+
 set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../.. CACHE INTERNAL "")
-set(TEST_TARGET mbed-drivers-buffered-serial)
+set(APP_TARGET mbed-drivers-buffered-serial)
 
 include(${MBED_PATH}/tools/cmake/mbed_greentea.cmake)
 
-project(${TEST_TARGET})
+project(${APP_TARGET})
 
-mbed_greentea_add_test(TEST_NAME ${TEST_TARGET})
+mbed_greentea_add_test(TEST_NAME ${APP_TARGET} TEST_REQUIRED_LIBS ${MBED_TEST_LINK_LIBRARIES})
```

Comment:
`TEST_REQUIRED_LIBS ` this can be removed, `MBED_TEST_LINK_LIBRARIES` should be used instead?